crates/router/src/connector/bambora/transformers.rs

```diff
@@ -0,0 +1,194 @@
+    fn try_from(item: &types::PaymentsAuthorizeRouterData) -> Result<Self,Self::Error> {
+        let payment_method_data = item.request.payment_method_data.clone();
+        let payment_method =
+            match payment_method_data {
+                api::PaymentMethod::Card(ref _item) => String::from("card"),
+                _ => todo!()
+            };
+        let card =
+            match payment_method_data {
+                api::PaymentMethod::Card(ref item) => {
+                    Card {
+                        name: item.card_holder_name.peek().clone(),
+                        number: item.card_number.peek().clone(),
+                        expiry_month: item.card_exp_month.peek().clone(),
+                        expiry_year: item.card_exp_year.peek().clone(),
+                        cvd: item.card_cvc.peek().clone(),
+                        complete: true
+                    }
+                }
+                _ => todo!()
+            };
+        Ok(Self {
+            amount: item.request.amount,
+            payment_method: payment_method,
+            card,
+        })
+    }
+}
+
+//TODO: Fill the struct with respective fields
+// Auth Struct
+pub struct BamboraAuthType {
+    pub(super) api_key: String
+}
+
+impl TryFrom<&types::ConnectorAuthType> for BamboraAuthType  {
+    type Error = error_stack::Report<errors::ConnectorError>;
+    fn try_from(_auth_type: &types::ConnectorAuthType) -> Result<Self, Self::Error> {
+        if let types::ConnectorAuthType::HeaderKey { api_key } = _auth_type {
+            Ok(Self {
+                api_key: api_key.to_string(),
+            })
+        } else {
+            Err(errors::ConnectorError::FailedToObtainAuthType.into())
+        }
+    }
+}
+// PaymentsResponse
+//TODO: Append the remaining status flags
+#[derive(Debug, Clone, Default, Serialize, Deserialize, PartialEq)]
+#[serde(rename_all = "lowercase")]
+pub enum BamboraPaymentStatus {
+    #[default]
+    ZERO,
+    ONE,
+}
+
+impl From<BamboraPaymentStatus> for enums::AttemptStatus {
+    fn from(item: BamboraPaymentStatus) -> Self {
+        match item {
+            BamboraPaymentStatus::ONE => Self::Charged,
+            BamboraPaymentStatus::ZERO => Self::Failure,
+        }
+    }
+}
+
+//TODO: Fill the struct with respective fields
+#[derive(Default, Debug, Clone, Serialize, Deserialize, PartialEq)]
+pub struct BamboraPaymentsResponse {
+    approved: BamboraPaymentStatus,
+    id: String,
+    response: Option<String>
+}
+
+impl<F,T> TryFrom<types::ResponseRouterData<F, BamboraPaymentsResponse, T, types::PaymentsResponseData>> for types::RouterData<F, T, types::PaymentsResponseData> {
+    type Error = error_stack::Report<errors::ParsingError>;
+    fn try_from(item: types::ResponseRouterData<F, BamboraPaymentsResponse, T, types::PaymentsResponseData>) -> Result<Self,Self::Error> {
+        Ok(Self {
+            status: enums::AttemptStatus::Charged,
```

Comment:
how are the handling the void flow here, where are the status mappings for void responses
Refer: https://github.com/juspay/hyperswitch/blob/e4956820fdfea8f637600b7fd03ac666f3b19c21/crates/router/src/connector/adyen/transformers.rs#L654

Reply:
According to the doc of Bambora all void payments get cancelled 
we just need to pass id and amount
<img width="1265" alt="Screenshot 2023-02-12 at 1 41 17 AM" src="https://user-images.githubusercontent.com/62378384/218279168-5826ca5c-289e-49fb-8573-7514b7c3324c.png">